vivisect/__init__.py

```diff
@@ -1122,8 +1128,12 @@ def parseOpcode(self, va, arch=envi.ARCH_DEFAULT):
             # so that at least parse opcode wont fail
             if loctup is not None and loctup[L_TINFO] and loctup[L_LTYPE] == LOC_OP:
                 arch = loctup[L_TINFO]
-
-        return self.imem_archs[(arch & envi.ARCH_MASK) >> 16].archParseOpcode(b, off, va)
+        key = (va, arch)
+        op = self.opcache.get(key)
+        if not op:
+            op = self.imem_archs[(arch & envi.ARCH_MASK) >> 16].archParseOpcode(b, off, va)
+            self.opcache[key] = op
+        return op
```

Comment:
this looks neat and id imagine it could really improve performance (at the expense of memory usage).

i'd like to see a way to clear this cache (is reaching into the `opcache` directly supported?), since the user (and self-modifying code) may change the underlying bytes.